octoprint_authldap/__init__.py

```diff
@@ -0,0 +1,271 @@
+		else:
+			return None
+
+	def findLDAPUser(self, userid):
+		userid = self.escapeLDAP(userid)
+		self._logger.debug("Searching User in ldap: %s" % userid)
+
+		connection = self.getLDAPClient()
+		try:
+			self._logger.debug("Binding LDAP with User: %s" % settings().get(["plugins", "authldap", "ldap_bind_user"]))
+			connection.simple_bind_s(
+				settings().get(["plugins", "authldap", "ldap_bind_user"]),
+				settings().get(["plugins", "authldap", "ldap_bind_password"])
+			)
+			query = settings().get(["plugins", "authldap", "ldap_query"]).format(uid=userid)
+			self._logger.debug("Searching for \"uid=%s\" under \"%s\" " % (query, settings().get(["plugins", "authldap", "ldap_search_base"])))
+			result = connection.search_s(
+				settings().get(["plugins", "authldap", "ldap_search_base"]),
+				ldap.SCOPE_SUBTREE,
+				query
+			)
+			connection.unbind_s()
+			self._logger.debug("Search finished.")
+			if result is None or len(result) == 0:
+				self._logger.debug("No User Found in LDAP")
+				return None
+			# Get the DN of first user found
+			dn, data = result[0]
+			self._logger.debug("User Found %s" % dn)
+			return dn
+		except ldap.SERVER_DOWN:
+			self._logger.debug("LDAP Server unreachable!")
+		return None
+
+	def checkPassword(self, username, password):
+		self._logger.debug("Checking Password")
+		if self.findLDAPUser(username) is not None:
+			self._logger.debug("User is a LDAP User")
+			try:
+				connection = self.getLDAPClient()
+				username = self.escapeLDAP(username)
+				dn = self.findLDAPUser(username)
+				connection.simple_bind_s(dn, password)
+				connection.unbind_s()
+				return True
+
+			except ldap.INVALID_CREDENTIALS:
+				self._logger.error("Username or password is incorrect.")
+		else:
+			self._logger.debug("User is a local user")
+			return self._localUserManager.checkPassword(username, password)
+
+	def changeUserPassword(self, username, password):
+		# Changing password of LDAP users is not allowed
+		if self.findLDAPUser(username) is None:
+			return self._localUserManager.changeUserPassword(username, password)
+		else:
+			self._logger.error("User is not allowed to change the Password.")
+
+	def getLDAPClient(self):
+		self._logger.debug("Creating LDAP Client")
+		ldap_server=settings().get(["plugins", "authldap", "ldap_uri"])
+		self._logger.debug("LDAP URL %s" % ldap_server)
+		if not ldap_server:
+			settings().remove(["accessControl", "userManager"])
+			self._logger.debug("UserManager: %s" % settings().get(["accessControl", "userManager"]))
+			raise Exception("LDAP conf error, server is missing")
+
+		connection = ldap.initialize(ldap_server)
+		connection.set_option(ldap.OPT_REFERRALS,0)
+		self._logger.debug("LDAP initialized")
+
+		if settings().get(["plugins", "authldap", "ldap_method"]) == 'TLS':
+			self._logger.debug("LDAP is using TLS, setting ldap options...")
+			ldap_verifypeer = settings().get(
+				["plugins", "authldap", "ldap_tls_reqcert"])
+			verifypeer = ldap.OPT_X_TLS_HARD
```

Comment:
Added all other kinds of TLS Verification.
If OPT_X_TLS_* will be directly written in the settings, the if/elif could be removed